### ISSUE TITLE
Expected performance for the AMI database

### ISSUE THREAD

**Issue by diego-fustes** [pyannote/pyannote-audio#122]
Hi folks. First of all, thanks for sharing this project, which looks very useful.
 I've followed the steps in the tutorial to train the speaker change detection on AMI. The resulting model achieved a loss of 0.23 in the training data. Furthermore, I've computed the metrics for the networks at several steps and the coverage is very low, from 13% down to 7%. Is this expected? Which performance do you get with this dataset?

Kind regards

**Comment by diego-fustes** [comment]
Sorry for my mistake, I did want to say speaker change detection :)

I'm trying to launch the evaluation with SegmentationCoverage and see how it turns out. I'll report the results here.

Thanks

**Comment by diego-fustes** [comment]
Indeed, now I get purity: 76 and coverage: 89 with the segmentation metrics. So, my question is, why is the tutorial using the diarization metrics? should it be updated?

**Comment by hbredin** [comment]
I'd happily merge a pull request adding a `--segmentation` option to the `pyannote-speaker-change validate` command line tool.

**Comment by diego-fustes** [comment]
Sorry for the long delay here. I'm preparing the pull request, but before sending it, I have a couple of questions:

1) Does it make sense to compute the diarization error in the case of change detection? Depending on the answer:

      - Yes: Looking at the code base, the only way to implement a condition for the scd validation is to add a parameter to the validate method in the base application, which would affect all applications. I would call it "metric_type" and would depict the metric to be used for validation. We would need to create a Factory that creates and object in function of this metric type and then assign names to the several metrics available for validation (e.g 'DiarizationPurityCoverage' or 'SegmentationPurityCoverage')

       - Not: we just need to update the validate_epoch method in change_detecion to apply the 
segmentation metrics. Nevertheless, I think that a metric_type parameter would benefit the framework... It's up to you

2) Following the tutorial, I do not see a validation routine for the diarization pipeline, which I think it's needed for the users (I'm developing it myself). Should we add it to the framework?

Kind regards

**Comment by hbredin** [comment]
Thanks a lot for willing to contribute new features to the project. That would be a first! 

Please use the `develop` branch as your base branch for any pull request: this is where most development happens before being released in the `master` branch. 

1. I think both diarization and segmentation metrics are useful but I agree that segmentation metrics are a better fit for speaker change detection. The reason why I used diarization metrics in the first place is because I wanted to evaluate the impact of each step of a diarization pipeline (and draw figures like the one available [here](http://pyannote.github.io/pyannote-metrics/reference.html#use-case))

I don't think there is a need for a global `metric_type` attribute common to all applications. You can directly set a `metric_type` attribute [in the same way it is already done for the `purity` attribute](https://github.com/pyannote/pyannote-audio/blob/develop/pyannote/audio/applications/change_detection.py#L325)

But I am open to suggestion, are there other applications where you think a global `metric_type` would  be really useful?

2. You refer to the `pyannote-pipeline` command line tool right? and the fact that it does not have a `validate` mode? Indeed, it could be useful. However, there are no notion of "epochs" for pipelines. How does your implementation work? It watches the content of `params.yml` and run an experiment every time it changes? That would be a nice addition to the framework indeed.

On a side note, I would like to create a separate project for pipeline hyperparameter optimisation in the future, called `pyannote.pipeline`. Indeed, there is no reason pipelines are limited to audio/speech processing and it would be useful to have it as a standalone library.

**Comment by diego-fustes** [comment]
1. Ok, so let's start with the implementation. Thanks to pointing to a way to set the `metric_type`. I would argue that it's not very elegant to set properties this way as it breaks encapsulation, but anyway. Before sending a pull request to change this repo, following the current implementation I need to add a new metric to pyannote-metrics, which is included in this [pull request](https://github.com/pyannote/pyannote-metrics/pull/23)

2. Yeah, I was thinking in something like that, but I need to clarify myself some parts of it before I can implement it. I'll let you know

Thanks

**Comment by diego-fustes** [comment]
Now that pyannote-metrics has been updated, I've created a [pull request](https://github.com/pyannote/pyannote-audio/pull/129) for this issue. Please check it out

**Comment by hbredin** [comment]
FYI, I just released a new package called [`pyannote.pipeline`](https://github.com/pyannote/pyannote-pipeline) which is a complete rewrite of the tunable pipelines features of pyannote.audio. It is still lacking a validation mode, though. 

`develop` branch of `pyannote.audio` has been updated to use this new package.

**Comment by sean90175** [comment]
HI
I've followed the tutorial on speaker change detection using AMI dataset and I've changed the evaluation metrics from Diarization metrics to segmentation metrics.
Howerver, I still can't get reasonable result no matter how I tune the alpha value.
Which parameters should I change in order to get better result?
Thank you.

**Comment by hbredin** [comment]
Not sure. How long (how many epochs) did you train the model?  
Which options did you use in "validate" mode (in particular --purity=???)?  
What performance do you get exactly according to "validate" mode?
